src/Microsoft.AspNet.Mvc.TagHelpers/InputTagHelper.cs

```diff
@@ -374,35 +386,38 @@ private static IEnumerable<string> GetInputTypeHints(ModelExplorer modelExplorer
                 }
```

Comment:
as mentioned in `TemplateRenderer`, suggest removing this code and using a shared helper.  (when I made my offline point about keeping the two methods' structure aligned, I didn't realize how much was identical)